zipline/pipeline/factors/statistical.py

```diff
@@ -475,3 +480,146 @@ def __new__(cls,
+    dtype = float64_dtype
+    params = ('allowed_missing_count',)
+
+    @expect_types(
+        regression_length=int,
+        target=Asset,
+        allowed_missing_percentage=(int, float),
+        __funcname='SimpleBeta',
+    )
+    @expect_bounded(
+        regression_length=(3, None),
+        allowed_missing_percentage=(0.0, 1.0),
+        __funcname='SimpleBeta',
+    )
+    def __new__(cls,
+                target,
+                regression_length,
+                allowed_missing_percentage=0.25):
+        daily_returns = Returns(
+            window_length=2,
+            mask=(AssetExists() | SingleAsset(asset=target)),
+        )
+        allowed_missing_count = int(np.floor(
+            allowed_missing_percentage * regression_length
+        ))
+        return super(SimpleBeta, cls).__new__(
+            cls,
+            inputs=[daily_returns, daily_returns[target]],
+            window_length=regression_length,
+            allowed_missing_count=allowed_missing_count,
+        )
+
+    def compute(self,
+                today,
+                assets,
+                out,
+                all_returns,
+                target_returns,
+                allowed_missing_count):
+        vectorized_beta(
+            dependents=all_returns,
+            independent=target_returns,
+            allowed_missing=allowed_missing_count,
+            out=out,
+        )
+
+    def __repr__(self):
+        return "{}(window_length={}, allowed_missing={})".format(
+            type(self).__name__,
+            self.window_length,
+            self.params['allowed_missing'],
+        )
+
+
+def vectorized_beta(dependents, independent, allowed_missing, out=None):
+    """
+    Compute slopes of linear regressions between columns of ``dependents`` and
+    ``independent``.
+
+    Parameters
+    ----------
+    dependents : np.array[N, M]
+        Array with columns of data to be regressed against ``independent``.
+    independent : np.array[N, 1]
+        Independent variable of the regression
+    allowed_missing : int
+        Number of allowed missing (NaN) observations per column. Columns with
+        more than this many non-nan observations in both ``dependents`` and
+        ``independents`` will output NaN as the regression coefficient.
+
+    Returns
+    -------
+    slopes : np.array[M]
+        Linear regression coefficients for each column of ``dependents``.
+    """
+    # Cache these as locals since we're going to call them multiple times.
+    nan = np.nan
+    isnan = np.isnan
+    N, M = dependents.shape
+
+    if out is None:
+        out = np.full(M, np.nan)
+
+    # Copy N times as a column vector and fill with nans to have the same
+    # missing value pattern as the dependent variable.
+    #
+    # PERF_TODO: We could probably avoid the space blowup by doing this in
+    # Cython.
+
+    # shape: (N, M)
+    independent = np.where(
+        isnan(dependents),
+        nan,
+        independent,
+    )
```

Comment:
Should you do a similar operation on the `dependent` matrix? Otherwise it will be using rows in its residual calculation that the independent variable does not use.

Reply:
Centering dependent turns out not to matter.  See 6bacafab65f8278a3730592b334fdd6dda953dea for a lengthy description.  (We discussed this in person, but putting the link here for posterity.)